### PR TITLE
Nested Queries for Wiki Edits

### PR DESCRIPTION
**Pull request by relwell**
This is mostly work from @Tonyynot14  and @micronix

I managed to get the agg filter query working by putting it into the search body, since using the existing agg filter affordances wasn't working. I'm sure we could get it working with the filter field, but it's a piece of tech debt I think we can afford to take for now.

This introduces the functionality to aggregate and filter against wiki page edits.

It also updates tests to accommodate general changes to how we use the query DSL to accommodate this level of filtering, and updates snapshots.

### CODE REVIEW

File: app/services/search_service.rb [hunk 1]
```diff
@@ -118,11 +135,11 @@ def search(search_after: nil, reverse: false, includes: [])
       &.map { |bucket| "fm_#{bucket[:key]}" } || []
 
     aggs = (crowd_aggs + ENABLED_AGGS).map { |agg| [agg, { limit: 10 }] }.to_h
-
     options = search_kick_query_options(aggs: aggs, search_after: search_after, reverse: reverse)
     options[:includes] = includes
     search_result = Study.search("*", options) do |body|
-      body[:query][:bool][:must] = { query_string: { query: search_query } }
+      body[:query][:bool][:must] = [{ query_string: { query: search_query } }]
```
Comment (relwell) [inline app/services/search_service.rb:141]: note the change from a hash to an array of hashes

File: db/schema.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2019_12_31_232856) do
+ActiveRecord::Schema.define(version: 2020_03_04_181505) do
```
Comment (relwell) [inline db/schema.rb:13]: not sure why this schema change is in this branch -- we need to verify this is correct.

File: docker-compose.yml [hunk 3]
```diff
@@ -1,89 +1,93 @@
...
+  elastic:
+    image: elasticsearch:6.8.1
+    container_name: elastic
+    ports:
+      - 9200:9200
+    environment:
+      - discovery.type=single-node
+      - http.port=9200
+      - http.cors.enabled=true
+      - http.cors.allow-origin=http://localhost:3030,http://127.0.0.1:3030,http://localhost:1358,http://127.0.0.1:1358,https://opensource.appbase.io
```
Comment (relwell) [inline docker-compose.yml:31]: this enables the usage of [mirage](https://github.com/appbaseio/mirage) and [deja vu](https://github.com/appbaseio/dejavu)

File: spec/services/search_service_spec.rb [hunk 4]
```diff
@@ -26,7 +24,7 @@
           headers: { "Content-Type" => "application/json" },
         )
       subject
-      expect(webmock_requests[0].body).to match_snapshot("default_query")
+      expect(JSON.pretty_generate(JSON.parse(webmock_requests.last.body))).to match_snapshot("default_query")
```
Comment (relwell) [inline spec/services/search_service_spec.rb:27]: this makes it easier to read the snapshot file

**Comment by williamhoos** [comment]
@Tonyynot14 not sure of status of the PR or branch but do you have everything you need from it to continue with nesting?

**Comment by williamhoos** [comment]
closing til ready

**Pull request closed without merging**